packages/dds/merge-tree/src/test/obliterateOperations.ts

```diff
@@ -0,0 +1,201 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+
+import { assert } from "node:console";
+
+import type { IRandom } from "@fluid-private/stochastic-test-utils";
+
+import type { IMergeTreeOp } from "../ops.js";
+import { InteriorSequencePlace, Side } from "../sequencePlace.js";
+
+import { annotateRange, type TestOperation } from "./mergeTreeOperationRunner.js";
+import type { TestClient } from "./testClient.js";
+
+export const posInField = (
+	client: TestClient,
+	pos: number,
+): { startPos: number; endPos: number } | undefined => {
+	if (
+		pos >= client.getLength() ||
+		(!Number.isInteger(Number(client.getText(pos, pos + 1))) &&
+			client.getText(pos, pos + 1) !== "{" &&
+			client.getText(pos, pos + 1) !== "}")
+	) {
+		return undefined;
+	}
+
+	let startPos = pos;
+	let endPos = pos;
+	// To find the start and end separators, walk backwards and forwards until the desired character is found.
+	while (
+		startPos > 0 &&
+		client.getText(startPos, startPos + 1) !== "{" &&
+		(client.getText(startPos, startPos + 1) === "}" ||
+			Number.isInteger(Number(client.getText(startPos, startPos + 1))))
+	) {
+		startPos--;
+	}
+
+	while (
+		endPos < client.getLength() &&
+		client.getText(endPos, endPos + 1) !== "}" &&
+		(client.getText(endPos, endPos + 1) === "{" ||
+			Number.isInteger(Number(client.getText(endPos, endPos + 1))))
+	) {
+		endPos++;
+	}
+
+	assert(startPos >= 0 && endPos <= client.getLength(), "Field endpoints are out of bounds");
```

Comment:
This isn't necessary considering the while conditions, unless pos can be negative.